app/src/main/java/acr/browser/lightning/view/LightningWebClient.kt

```diff
@@ -105,23 +104,22 @@ class LightningWebClient(
     }
 
     private fun chooseAdBlocker(): AdBlocker = if (userPreferences.adBlockEnabled) {
-        activity.injector.provideBloomFilterAdBlocker()
+        activity.injector.provideAbpAdBlocker()
```

Comment:
I still haven't compared performance of Bloom Filer vs the new blocker.
If it's considerably better, then point 2 and 3 of _Ideas for changes_ could be implemented. Then providing hosts lists should result in nearly same blocking speed as the old blocker.

Reply:
So in my opinion we should only have one blocker, but internally it might be some combination of Bloom filter and the new blocker.

Reply:
At the moment my preference would be to have an architecture that supports multiple technology Adblock, Adblock Plus and uBlock Origin whatever comes tomorrow. At least at the developer level it should be easy to switch between them and ideally directly from the user settings. We already have the basic architecture in place so it should be fairly easy to preserve and extend it as needed.

Reply:
I did some block speed test, using the same list of hosts for the new Yuzu-based blocker and the Bloom filter blocker:
Bloom filter takes about 100 µs per request, while Yuzu takes 260 µs per request.
However, the Yuzu blocker also needs to extract more data from the WebResourceRequest (not for hosts-list, but for other block rules). If this is considered, it takes 350 µs per block with 3rd party cache, and 650 µs per block without.

So for pure hosts lists there is a clear advantage for the Bloom filter (100 µs vs 260 µs, factor of 2.6).
But if we also have other blocking rules, extracting the additional data from the request is unavoidable, which decreases the advantage of Bloom filter to 190 µs vs 360 µs (factor 1.9).

If the Bloom filter lands a hit, it's actually a maybe-hit and needs to be checked/confirmed by Yuzu blocker anyway. Since there are many `tag` hits in Yuzu blocker that don't match due to other parts of the rules, I think full integration of the Bloom filter (for `tag` search) will give us little time save.

However, it might be reasonable to extract the hosts-rules from yuzu and check them using the Bloom filter. How useful this is depends of course on the used lists.

Regarding the technology: AdBlock Plus and uBlock Origin basically use the same filter rules (with some additonal rules for uBlock Origin), so with this filter we're supporting both anyway.